webapp/src/components/emptyCenterPanel.tsx

```diff
@@ -1,34 +1,170 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
-import React from 'react'
-import {FormattedMessage} from 'react-intl'
+import React, {useCallback, useEffect} from 'react'
+import {FormattedMessage, useIntl} from 'react-intl'
+import {generatePath, useHistory, useRouteMatch} from 'react-router-dom'
 
 import {getCurrentWorkspace} from '../store/workspace'
-import {useAppSelector} from '../store/hooks'
+import {useAppSelector, useAppDispatch} from '../store/hooks'
+import {Utils} from '../utils'
+import {Board} from '../blocks/board'
+import {getGlobalTemplates, fetchGlobalTemplates} from '../store/globalTemplates'
+import {getSortedTemplates} from '../store/boards'
+import AddIcon from '../widgets/icons/add'
+import BoardIcon from '../widgets/icons/board'
+import octoClient from '../octoClient'
+
+import {addBoardTemplateClicked, addBoardClicked} from './sidebar/sidebarAddBoardMenu'
+import {addBoardFromTemplate, BoardTemplateMenuItemOptionsButton} from './sidebar/boardTemplateMenuItem'
+
 import './emptyCenterPanel.scss'
 
-const EmptyCenterPanel = React.memo(() => {
-    const workspace = useAppSelector(getCurrentWorkspace)
+type ButtonProps = {
+    buttonIcon: string | React.ReactNode,
+    title: string,
+    readonly: boolean,
+    onClick: () => void,
+    showBoard?: (boardId: string) => void
+    boardTemplate?: Board
+}
+
+const EmptyCenterPanelButton = React.memo((props: ButtonProps) => {
+    const {onClick, buttonIcon, title, readonly, showBoard, boardTemplate} = props
 
     return (
-        <div className='EmptyCenterPanel'>
-            {workspace && workspace.id !== '0' &&
-            <div className='WorkspaceInfo'>
-                <FormattedMessage
-                    id='EmptyCenterPanel.workspace'
-                    defaultMessage='This is the workspace for:'
+        <div
+            onClick={onClick}
+            className='button'
+        >
+            <i>{buttonIcon}</i>
+            <span className='button-title'>{title}</span>
+            {!readonly &&
+                <BoardTemplateMenuItemOptionsButton
+                    showBoard={showBoard!}
+                    boardTemplate={boardTemplate!}
```

Comment:
I find this to be very confusing. The "!" tells the compiler that this variable is NOT null or undefined, yet there is no other check to verify that and there are definitely instances these are undefined.  I'm actually surprised this works, I had to try it.  I guess the React simply doesn't render if a required property is null.

In any case, it may be better explicitly checking for those variables first. 

```
{!readonly && showBoard && boardTemplate &&
     <BoardTemplateMenuItemOptionsButton
     ...
```